src/features/vault.ts

```diff
@@ -281,3 +284,19 @@ const execCwd = (cmd: string, cwd: string | undefined) => {
   }
   return exec(cmd, { cwd: cwd });
 };
+
+const getIndentationLevel = (
+  editor: vscode.TextEditor,
+  selection: vscode.Selection
+): number => {
+  if (!editor.options.tabSize) {
+    // according to VS code docs, tabSize is always defined when getting options of an editor
+    throw 'tabSize undefined, this should never happen';
+  }
+  const startLine = editor.document.lineAt(selection.start.line).text;
+  const indentationMatches = startLine.match(/^\s*/);
+  const leadingWhitespaces = !!indentationMatches
+    ? indentationMatches[0].length
+    : 0;
```

Comment:
Could this be something like
```suggestion
  const leadingWhitespaces = indentationMatches?[0].?length || 0;
```
?